### ISSUE TITLE
Broken html-check-links

### ISSUE THREAD

**Issue by d-torrance** [Macaulay2/M2#2398]
There are currently two issues with the `html-check-links` utility:

* Currently, `make -C Macaulay2/html-check-links` only checks xhtml files, of which there are none.  This was because of a mistake I made in [#1875](https://github.com/Macaulay2/M2/pull/1875).
* After restoring the original behavior of the check, we get the following on the autotools GitHub builds, e.g., [this one](https://github.com/Macaulay2/M2/runs/5347144612?check_suite_focus=true).  I can't reproduce this locally, and I don't understand why it thinks that `YYSTYPE` is `int`.  That's what it's initially defined as in `grammar-tab.h` by bison, but it's supposed to be redefined as `char *` in `grammar.h`.  Why isn't it?
```
gcc -std=gnu11 -g3 -O2 -Wall -g -I/home/runner/work/M2/M2/M2/BUILD/build/../../include -I/home/runner/work/M2/M2/M2/BUILD/build/include -I/home/runner/work/M2/M2/M2/BUILD/build/usr-host/include -isystem /usr/include/libxml2     -I/usr/src/gtest  -I/home/linuxbrew/.linuxbrew/include -I/home/linuxbrew/.linuxbrew/opt/tbb@2020/include -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include/python3.8 -I/usr/include -DBOOST_STACKTRACE_LINK -I../../../../Macaulay2/html-check-links -I.  -c grammar.fixed.c -o grammar.fixed.o
      |               ^
../../../../Macaulay2/html-check-links/lex.l:38:15: warning: assignment to ‘YYSTYPE’ {aka ‘int’} from ‘char *’ makes integer from pointer without a cast [-Wint-conversion]
   38 | <tag>src   COUNT; yylval = strperm(yytext); return SRC;
      |               ^
../../../../Macaulay2/html-check-links/lex.l:39:16: warning: assignment to ‘YYSTYPE’ {aka ‘int’} from ‘char *’ makes integer from pointer without a cast [-Wint-conversion]
   39 | <tag>{STRING}   COUNTN; yylval = strperm(yytext); return WORD;
      |                ^
../../../../Macaulay2/html-check-links/lex.l:40:16: warning: assignment to ‘YYSTYPE’ {aka ‘int’} from ‘char *’ makes integer from pointer without a cast [-Wint-conversion]
   40 | <tag>{WORD}   COUNTN; yylval = strperm(yytext); return WORD;
      |                ^
../../../../Macaulay2/html-check-links/grammar.y: In function ‘yyparse’:
../../../../Macaulay2/html-check-links/grammar.y:52:36: warning: passing argument 1 of ‘checkURL’ makes pointer from integer without a cast [-Wint-conversion]
   52 |     | key '=' WORD { checkURL($3); }
      |                               ~~~~~^  
      |                                    |
      |                                    YYSTYPE {aka int}
gcc -L/home/runner/work/M2/M2/M2/BUILD/build/usr-host/lib -L/home/linuxbrew/.linuxbrew/lib     -L/home/linuxbrew/.linuxbrew/opt/tbb@2020/lib   -L/home/linuxbrew/.linuxbrew/opt/python@3.9/Frameworks/Python.framework/Versions/3.8/lib -g3 -L/usr/lib/x86_64-linux-gnu  html-check-links.o getmem.o lex.o grammar.fixed.o -lgc  -o html-check-links
../../../../Macaulay2/html-check-links/grammar.y:33:22: note: expected ‘char *’ but argument is of type ‘YYSTYPE’ {aka ‘int’}
   33 | static void checkURL(char *);
      |                      ^~~~~~
xargs: /home/runner/work/M2/M2/M2/BUILD/build/Macaulay2/html-check-links/html-check-links: terminated by signal 11
```

**Comment by DanGrayson** [comment]
I can't reproduce it here, either.  As far as I can see, the file grammar.tab.h is not included when compiling the two c files.

**Comment by d-torrance** [comment]
Fixed in #3303

**Comment by d-torrance** [comment]
Reopening -- the second bullet point above is still an issue on RHEL 9 (see https://github.com/d-torrance/M2-workflows/actions/runs/11410500788/job/31752935838)

**Comment by d-torrance** [comment]
Closed (again!) by #3528
